atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/ConnectionConfig.java

```diff
@@ -123,6 +123,8 @@ public HikariConfig getHikariConfig() {
         config.setMaxLifetime(TimeUnit.SECONDS.toMillis(getMaxConnectionAge()));
         config.setIdleTimeout(TimeUnit.SECONDS.toMillis(getMaxIdleTime()));
         config.setLeakDetectionThreshold(getUnreturnedConnectionTimeout());
+        // Not a bug - we don't want to use connectionTimeout here, since Hikari uses a different terminology.
+        // See https://github.com/brettwooldridge/HikariCP/wiki/Configuration - connectionTimeout = how long to wait for a connection to be opened.
```

Comment:
Might be worth noting that the actual socket timeouts from `ConnectionConfig` `connectionTimeoutSeconds` and `socketTimeoutSeconds` are mapped in `com.palantir.nexus.db.pool.config.PostgresConnectionConfig#getHikariProperties` and `com.palantir.nexus.db.pool.config.OracleConnectionConfig#getHikariProperties` respectively.